dashi/poolqc/pooling_qc_sample.py

```diff
@@ -132,74 +210,71 @@ def update_title(lane_value, run_value):
-         'Proportion Intronic Bases', 'Proportion Intergenic Bases',
-         'rRNA Contamination (%reads aligned)', 'Proportion Correct Strand Reads',
+        ['library', 'Run', 'LaneNumber', 'Index1', 'Index2', 'Clusters', 'rRNA Contamination (%reads aligned)',
+         'Proportion Coding Bases',
+         'Proportion Intronic Bases',
+         'Proportion Intergenic Bases', 'Proportion Correct Strand Reads',
          'Coverage'])
     run = run.sort_values('library')
+
     csv = run.to_csv(index=False)
     csv = 'data:text/csv;charset=utf-8,' + urllib.parse.quote(csv)
-    run = run.drop(columns=['Run', 'LaneNumber'])
 
-    # Append Pass/Fail and Thresholds
+    run = run.drop(columns=['Run', 'LaneNumber'])
+    run = run.dropna(axis=1, how='all', thresh=None, subset=None, inplace=False)
 
     columns = [{'name': i, 'id': i,
                 'type': 'numeric'} for i in run.columns]
 
     data = run.to_dict('records')
 
-    # highlighting datatable cells/columns/rows
-    style_data_conditional = [{
-        'if': {'column_id': 'library'},
-        'backgroundColor': 'rgb(222, 222, 222)'
-    },
-        {'if': {'column_id': 'properly paired reads',
-                'filter_query': '0 < {properly paired reads} and {properly paired reads} < 20'},
-         'backgroundColor': 'rgb(219, 75, 75)'},
-        {'if': {'column_id': 'Proportion Coding Bases',
-                'filter_query': '0 < {Proportion Coding Bases} and {Proportion Coding Bases} < 20'},
-         'backgroundColor': 'rgb(219, 75, 75)'},
-        {'if': {'column_id': 'Proportion Intronic Bases',
-                'filter_query': '0 < {Proportion Intronic Bases} and {Proportion Intronic Bases} < 20'},
-         'backgroundColor': 'rgb(219, 75, 75)'},
-        {'if': {'column_id': 'Proportion Intergenic Bases',
-                'filter_query': '0 < {Proportion Intergenic Bases} and {Proportion Intergenic Bases} < 15'},
-         'backgroundColor': 'rgb(219, 75, 75)'},
-        {'if': {'column_id': 'rRNA Contamination (%reads aligned)',
-                'filter_query': '0 < {rRNA Contamination (%reads aligned)} and {rRNA Contamination (%reads aligned)} < 15'},
-         'backgroundColor': 'rgb(219, 75, 75)'},
-    ]
-    return columns, data, style_data_conditional, csv
-
-
-def update_sampleindices(run):
+    return columns, data, csv
+
+
+def update_sampleindices(run, index_threshold):
     run = run.sort_values('library')
-    num_libraries = len(run['library'])
-    samples_passing_clusters = '%s/%s' % (sum(i > INDEX_THRESHOLD for i in run['SampleNumberReads']), num_libraries)
 
     data = []
 
     for inx, d in run.groupby(['library']):
-        d['Threshold'] = INDEX_THRESHOLD
-        d['Color'] = np.where((d['SampleNumberReads'] >= INDEX_THRESHOLD), '#ffffff', '#db4b4b')
+        d['Threshold'] = index_threshold
+        d['Color'] = np.where((d['Clusters'] >= index_threshold), '#20639B', '#db4b4b')
```

Comment:
I get the following warning here:

```
/home/slazic/Documents/Code/2018-08-27-Dash_trial/dashi/poolqc/pooling_qc_sample.py:270: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy

/home/slazic/Documents/Code/2018-08-27-Dash_trial/dashi/poolqc/pooling_qc_sample.py:271: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead
```

You are potentially changing the global `df` variable, which would cause big issues if two or more people load this app.

Reply:
I'm not getting this error, I''ve tried running the app via `index.py` as well as `pooling_qc_sample.py`

Reply:
@slazicoicr I'm also not seeing this error. Was there a particular run which was causing this error?

However, I do think it's worth looking into Savo's concern about whether the global `df` variable is being changed.

Reply:
I see it on startup in the terminal when I run `python dashi/poolqc/pooling_qc_sample.py`. It might be due to version issue. I am on Pandas 0.23.4